src/sniffer/pkg/sniffer/sniffer.go

```diff
@@ -14,23 +13,27 @@ import (
 	"time"
 )
 
+// capturesMap is a map of source IP to a map of destination DNS to the last time it was seen
+type capturesMap map[string]map[string]time.Time
+
 type Sniffer struct {
-	capturedRequests map[string]*goset.Set[string]
+	capturedRequests capturesMap
```

Comment:
As discussed: please type def the individual types so the map itself is meaningful, but not the entire thing (capturesMap is just as meaningful as the name, capturedRequests, but doesn't say anything about the structure of what's in the map)

For example: map[SourceIP]map[DNSName]time.Time